deeplink/build.gradle.kts

```diff
@@ -10,37 +16,96 @@ android {
         minSdkVersion(Android.minSdk)
         targetSdkVersion(Android.sdk)
         versionName = Project.version
-        setProperty("archivesBaseName", Project.name)
+        setProperty("archivesBaseName", "${Project.name}-${Project.version}")
         testInstrumentationRunner = "android.support.test.runner.AndroidJUnitRunner"
     }
 }
 
 group = Project.groupId
 version = Project.version
 
-val artifact = "$buildDir/outputs/aar/${Project.name}-${Project.version}.aar"
+dependencies {
+    implementation(Dependencies.kotlinStdLib)
+    implementation(Dependencies.okio)
+    testImplementation(DependenciesTest.junit)
+    testImplementation(DependenciesTest.kotlinTest)
+    androidTestImplementation(DependenciesTest.junit)
+    androidTestImplementation(DependenciesTest.kotlinTest)
+    androidTestImplementation(DependenciesTest.supportTestRunner)
+}
+
+fun MavenPom.addDependencies() = withXml {
+    asNode().appendNode("dependencies").let { depNode ->
+        configurations.compile.allDependencies.forEach {
+            depNode.appendNode("dependency").apply {
+                appendNode("groupId", it.group)
+                appendNode("artifactId", it.name)
+                appendNode("version", it.version)
+            }
+        }
+    }
+}
+
+val artifact = "$buildDir/outputs/aar/${Project.name}-${Project.version}-release.aar"
+
+val sourcesJar by tasks.creating(Jar::class) {
+    classifier = "sources"
+    from(android.sourceSets.getByName("main").java.srcDirs)
+}
+
+val dokka by tasks.getting(DokkaTask::class) {
+    outputFormat = "html"
+    outputDirectory = "$buildDir/javadoc"
+}
+
+val dokkaJar by tasks.creating(Jar::class) {
+    group = JavaBasePlugin.DOCUMENTATION_GROUP
+    description = "Assembles Kotlin docs with Dokka"
+    classifier = "javadoc"
+    from(dokka)
+}
+
+artifacts {
+    add("archives", sourcesJar)
+    add("archives", dokkaJar)
+}
+
+
 publishing {
 
     publications {
         register(Project.artifactId, MavenPublication::class) {
             groupId = Project.groupId
             artifactId = Project.artifactId
-            artifact(artifact)
             version = Project.version
+            artifact(artifact)
+            artifact(sourcesJar)
+            artifact(dokkaJar)
+            pom.addDependencies()
         }
     }
 }
 
-dependencies {
-    implementation(Dependencies.kotlinStdLib)
-    implementation(Dependencies.okio)
-    testImplementation(DependenciesTest.junit)
-    testImplementation(DependenciesTest.kotlinTest)
-    androidTestImplementation(DependenciesTest.junit)
-    androidTestImplementation(DependenciesTest.kotlinTest)
-    androidTestImplementation(DependenciesTest.supportTestRunner)
-}
-
-repositories {
-    mavenCentral()
+bintray {
+    user = System.getenv("BINTRAY_USER") ?: ""
+    key = System.getenv("BINTRAY_API_KEY") ?: ""
+    publish = true
+    setPublications(Project.artifactId)
+    with(pkg) {
```

Comment:
Could we use `pkg(closureOf<PackageConfig> { ... })` as is used [here](https://docs.gradle.org/5.0/userguide/kotlin_dsl.html#groovy_closures_from_kotlin) ?

Reply:
I see. I had it like this earlier but thought the `with` function makes it a shorthand. No?

Reply:
Yea. Let's leave it as is then